### PR TITLE
[Feat]: Tag grouping by translation

### PR DESCRIPTION
**Pull request by EnergoStalin**
Will fix "Tag with slightly different kanji and exact same translation" from #93

### CODE REVIEW

**Comment by EnergoStalin** [comment]
Need to add some UI toggles for tag grouping. Likely wont do it myself.

File: web/vview/widgets/tag-search-dropdown.js [hunk 1]
```diff
@@ -1032,6 +1032,83 @@ class TagSearchDropdownWidget extends widget
...
+                    slug = Object.keys(acc.groups).find(key => slug.startsWith(key)) ?? slug;
+                    slug = slug.split(/[ _-]/g)[0]
+                }
+            }
+
+            if (!acc.groups[slug]) {
+                acc.groups[slug] = {
+                    tag: new Set([tag.tag]),
+                    // Downside of this approach is that joined tag list shoud be inserted in fixed place since we dont know position
+                    search: tag.search.replace(tag.tag, '')
```
Comment (EnergoStalin) [inline web/vview/widgets/tag-search-dropdown.js:1065]: Probably need to reparse  boundaries and store it for later insert position inference will fix later

**Comment by ppixiv** [comment]
Sorry, I spent some time brainstorming #93 but got distracted by other things and it fell off my radar.  I've pushed this with some refactoring.  Handling tags in the middle of the search works now, and there's a toggle in the section header.

For now I removed the unused options to avoid adding unused code, I can look at them later but I'm not sure how they're meant to work.  I'm guessing they're experimental since there was no way to turn them on.  I left them in commit history so it's easy for you to bring them back in on your end (just revert 5f649472).  Maybe it could just merge based on the string with delimiters removed?  That would merge "red dog" with "red_dog", but not "red house".  I haven't used it enough yet to know how useful that would be, though, I'm not seeing a lot of results that look like more aggressive merging would be helpful.

**Comment by EnergoStalin** [comment]
>  That would merge "red dog" with "red_dog", but not "red house"

It kinda does that but with things in brackets this already would be handled by JoinMonotags i belive.

> For now I removed the unused options

They meant to be used i just don't want write ui toggles for them. It put behind toggle cause it **might** be useful but not necessarily. General purpose of it is to allow more general grouping and not only by exact match of translation with brackets and symbols stripped.

## Use cases
The best way to understand use cases is to try yourself. I quite struggle to express most useful ones. Cause there's a lot different tag families try `feet` for example there's a lot inconvenient tags here which wont be grouped at all without flags with **JoinPrefixes** and **JoinMonotags** they all will be in `feet` group.

## JoinPrefixes
JoinPrefixes should join all groups with same prefix with shortest one.
So when input is `blue` it would join all groups which name starting with blue together. Might be useful sometimes i think.

### Example
```yaml
query: blue
JoinPrefixes: true
JoinMonotags: false
topmostGroup: ( 青 OR ブルー OR ブルー(トレーナー) OR ブルーアーカイブ OR 蔚蓝档案 ) 
```

## JoinMonotags
JoinMonotags will try to group tags without spaces but with same prefix might be useful if we don't want join by group prefix i guess?

### Example
```yaml
query: blue
JoinPrefixes: false
JoinMonotags: true
topmostGroup: ( ブルーアーカイブ OR 蔚蓝档案 OR BlueArchive )
```

## Both

### Example
```yaml
query: blue
JoinPrefixes: true
JoinMonotags: true
topmostGroup: ( 青 OR ブルー OR ブルー(トレーナー) OR ブルーアーカイブ OR 蔚蓝档案 OR BlueArchive ) 
```

**Comment by ppixiv** [comment]
merged via e01c54dfd5

**Pull request closed without merging**